spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SqlSpannerQuery.java

```diff
@@ -58,19 +63,32 @@ public SqlSpannerQuery(Class type, QueryMethod queryMethod,
 		this.tags = getTags();
 		this.evaluationContextProvider = evaluationContextProvider;
 		this.expressionParser = expressionParser;
-		this.sql = sql;
+		this.sql = StringUtils.trimTrailingCharacter(sql.trim(), ';');
+		if (this.sql.contains(";")) {
```

Comment:
What if the query has a where clause like:
```
where sybol = ';'
```
This seems pretty fragile. What does Spring Data JDBC do in this scenario?

Reply:
ah i keep forgetting that. I think we just shouldn't have that check there.

Reply:
I'm not sure if this appears in spring-data-jdbc. In any case, with Spanner client lib we must specify the, limit, offset, and sort for queries in the SQL string.